### DISCUSSION TITLE
Expand, Roles and mails

### DISCUSSION THREAD

**Discussion by Helixar** [pocketbase/pocketbase#1260] [Q&A] [answered]
Hello,
would it be possible to add the "expand" functionality to collections when using the subscribe function in real time?

For example :
if I use the "projects" collection I can also get the "teams" relationship
`const record = await client.collection('projects').getFirstListItem('myId', {expand: 'teams'})`

And then I want to retrieve the data in real time
`await client.collection('projects').subscribe('myId', (event) => {})`

The data I had retrieved via my relations is not transmitted via the subscribe method.

It would also be interesting to add a role system because for the moment if a user understands the system he can easily modify his role via a simple post request.

Another interesting idea would be to be able to send personalized mails. This would allow to create very powerful applications. This should be possible via a webhook system with a security token generated before.

With all these features it would be easy to create large applications.

Thank you for developing this project, it helps me a lot in the realization of my ideas.

**Comment by ganigeorgiev** [comment, marked as answer]
> would it be possible to add the "expand" functionality to collections when using the subscribe function in real time?

No, this is not possible at the moment. Eventually we may add support for it together with the server-side `filter` for the realtime events, but for now you'll have to send a separate `getOne`  request with your `expand` criteria.

---

> It would also be interesting to add a role system because for the moment if a user understands the system he can easily modify his role via a simple post request.

Roles for admins is in my to do to research it in the future since a lot of users requested it, but app user roles are not planned since this is already achievable using a custom field and an API rule (you can find examples with the different possible app user access controls in https://pocketbase.io/docs/collections/#auth-collection).

---

> Another interesting idea would be to be able to send personalized mails. This would allow to create very powerful applications. This should be possible via a webhook system with a security token generated before.

This is part of https://github.com/pocketbase/pocketbase/issues/317.
(_eventually the issue may get closed in favor of allowing binding to the [app event hooks](https://pocketbase.io/docs/event-hooks/) with JS, but that will be handled after v0.11/v0.12_)
